comms/storage/telemetry/tracer.go

```diff
@@ -0,0 +1,71 @@
+package telemetry
+
+import (
```

Comment:
A lot of this stuff I can see eventually moving to `shared` but for now it's here